### PR TITLE
Powershell implementation for Sql Server Dns Aliasing

### PR DESCRIPTION
**Pull request by maboja-msft**
The main aim of this pull request is to add Powershell implementation for Server Dns Aliasing feature for Azure Sql Server. The following cmdlets are implemented:
- Get-AzureRmSqlServerDnsAlias cmdlet which gets server dns aliases by server and alias name or a list of server dns aliases for an azure Sql Server.
 - New-AzureRmSqlServerDnsAlias cmdlet which creates new server dns alias for a given Azure Sql server
- Set-AzurermSqlServerDnsAlias cmlet which allows updating a Azure Sql Server to which server dns alias is pointing
- Remove-AzureRmSqlServerDnsAlias cmdlet which removes a server dns alias for a Azure Sql Serve

This change is also fixing a missing comma in GetAzureServiceBusRule implementation which was causing build to fail.

Since the newest version of .NET client dll for Sql is not published yet, I am for now  using local version. Will remove it from pull request once official nuget package for .NET SDK got published.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the new server to which alias should point
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The name of Azure Sql Server to which alias should point.")]
+		[ValidateNotNullOrEmpty]
+		public string TargetServerName { get; set; }
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: shouldn't there be a TargetServerResourceGroupName param? and maybe TargetServerSubscriptionId?

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+		/// <summary>
+		/// Persist changes - actually invoke acquire on alias to move it from one server to another
+		/// </summary>
+		protected override IEnumerable<Model.AzureSqlServerDnsAliasModel> PersistChanges(IEnumerable<Model.AzureSqlServerDnsAliasModel> entity)
+		{
+			return new List<Model.AzureSqlServerDnsAliasModel>()
+			{
+				ModelAdapter.AcquireServerDnsAlias(
+					this.ResourceGroupName,
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: oh I see, there is base .ResourceGroupName. Seems pretty confusing that this goes with TargetServerName.

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Model/AzureSqlServerDnsAliasModel.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		/// <summary>
+		/// Gets or sets the name of the server
+		/// </summary>
+		public string ServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of server dns alias
+		/// </summary>
+		public string DnsAliasName { get; set; }
+
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Model/AzureSqlServerDnsAliasModel.cs]: isn't there also a fully qualified dns name property?

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj [hunk 4]
```diff
@@ -262,6 +263,7 @@
     <None Include="ScenarioTests\DatabaseCrudStretchTests.ps1">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="ScenarioTests\ServerDnsAliasTests.ps1" />
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj]: @maboja-msft make sure this file gets copied to the output directory

File: src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj [hunk 5]
```diff
@@ -446,8 +454,8 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Sql.1.3.0-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>-->
     <Reference Include="Microsoft.Azure.Management.Sql, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Sql.1.7.0-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>
-      <Private>True</Private>
+      <HintPath>..\..\..\..\tools\LocalFeed\Microsoft.Azure.Management.Sql.1.7.0-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj]: @maboja-msft just a reminder to point at version 1.8.0-preview when it is published to NuGet

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/GetAzureSqlServerDnsAlias.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+	public class GetAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server DNS Alias
+		/// </summary>
+		[Parameter(Mandatory = false,
+			HelpMessage = "The Azure Sql Server DNS Alias name.")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/GetAzureSqlServerDnsAlias.cs]: @maboja-msft the naming should be swapped - the parameter name should be `Name` and the alias should be `DnsAliasName` (this is for consistency with other cmdlets)

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/NewAzureSqlServerDnsAlias.cs [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	/// <summary>
+	/// Defines the New-AzureRmSqlServerDnsAlias cmdlet
+	/// </summary>
+	[Cmdlet(VerbsCommon.New, "AzureRmSqlServerDnsAlias",
+ConfirmImpact = ConfirmImpact.Low, SupportsShouldProcess = true)]
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/NewAzureSqlServerDnsAlias.cs]: @maboja-msft please remove the `ConfirmImpact` setting

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/NewAzureSqlServerDnsAlias.cs [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	public class NewAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server Dns Alias
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The Azure Sql Server Dns Alias name.")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/NewAzureSqlServerDnsAlias.cs]: @maboja-msft same comment about swapping name and alias

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	[Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerDnsAlias", SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.Medium)]
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft same comment about removing `ConfirmImpact`

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+		/// <summary>
+		/// Gets or sets the name of the server dns alias to remove
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletParametersParameterSet,
+			Mandatory = true,
+			HelpMessage = "Azure Sql Server Dns Alias name")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft same comment about swapping name and alias

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+			HelpMessage = "The Azure Sql Server name.")]
+		[ValidateNotNullOrEmpty]
+		public override string ServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletParametersParameterSet,
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft please remove this line - `ResourceId` is the only parameter that should accept its value from the pipeline by property name

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+using System;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	/// <summary>
+	/// Defines the Set-AzureRmSqlServerDnsAlias cmdlet
+	/// </summary>
+	[Cmdlet(VerbsCommon.Set, "AzureRmSqlServerDnsAlias",
+		SupportsShouldProcess = true,
+ConfirmImpact = ConfirmImpact.Medium)]
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: @maboja-msft same comment about removing `ConfirmImpact`

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+		public static string ServerDnsAliasIdTemplate = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Sql/servers/{2}/dnsAliases/{3}";
+
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server Dns Alias
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The Azure Sql Server Dns Alias name.")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: @maboja-msft same comment about swapping name and alias

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+			ModelAdapter.RemoveServerDnsAlias(this.ResourceGroupName, this.ServerName, this.DnsAliasName);
+			return entity;
+		}
+
+		/// <summary>
+		/// Entry point for the cmdlet
+		/// </summary>
+		public override void ExecuteCmdlet()
+		{
+			if (!Force.IsPresent && !ShouldProcess(
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft at the design review, I think we talked about `Force` being used for the `ShouldContinue` prompt, but I don't see `ShouldContinue` in this cmdlet. Is it executed elsewhere? If not, you can see how it is used in this section on [`SupportsShouldProcess`](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).

File: src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+external help file: Microsoft.Azure.Commands.Sql.dll-Help.xml
+Module Name: AzureRM.Sql
+online version: 
+schema: 2.0.0
+---
+
+# Get-AzureRmSqlServerDnsAlias
+
+## SYNOPSIS
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md]: @maboja-msft please make sure that these templates (synopsis, description, etc.) are filled out for all of the markdown files you are adding

File: src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+```
+
+## DESCRIPTION
+{{Fill in the Description}}
+
+## EXAMPLES
+
+### Example 1
+```
+PS C:\> $serverDNSAliases = Get-AzureRmSqlServerDNSAlias -ServerName server -ResourceGroupName rg
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md]: @maboja-msft for the examples you have in your markdown files, please add the corresponding output the user will see when executing the command. This gives the user a better idea of what to expect from the object being returned to them from the cmdlet.

File: tools/StaticAnalysis/SignatureVerifier/CmdletSignatureMetadata.cs [hunk 17]
```diff
@@ -163,7 +163,8 @@ private static List<string> GetApprovedVerbs()
             "Statistics",
             "Status",
             "Success",
-            "Vmss"
+            "Vmss",
+            "Alias"
```
Comment (cormacpayne) [inline tools/StaticAnalysis/SignatureVerifier/CmdletSignatureMetadata.cs]: @maboja-msft minor detail, but can we keep these nouns in alphabetical order?

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Model/AzureSqlServerDnsAliasModel.cs (hunk 3 above)
Comment (maboja-msft) [reply src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Model/AzureSqlServerDnsAliasModel.cs]: This is basically the same thing like name, so I don't think we have to expose it

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs (hunk 2 above)
Comment (maboja-msft) [reply src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: I have added alias TargetResourceGroupName for ResourceGroupName

**Comment by maboja-msft** [comment]
Addressed comments, @cormacpayne please take a look. I will investigate why build failed.

**Comment by cormacpayne** [comment]
@maboja-msft we are having issues with Jenkins, so all builds are currently failing. I'll restart the job for your PR once we resolve the issues with Jenkins

**Comment by cormacpayne** [comment]
@maboja-msft it looks like the json recordings for your tests weren't checked in

File: src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+Aliases: AzureRmContext, AzureCredential
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -DnsAliasName
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md]: @maboja-msft looks like the help needs to be updated to reflect the changes that were made in the previous commit(s)

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	[Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerDnsAlias", SupportsShouldProcess = true)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class RemoveAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		protected const string CmdletParametersParameterSet =
+			"Remove a Server Dns Alias from cmdlet input parameters";
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft please make sure all parameter set names are singular strings (pascal case strings are the norm for parameter sets).

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/55/

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/72

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	[Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerDnsAlias", SupportsShouldProcess = true)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class RemoveAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		protected const string CmdletInputParameterSet =
+			"Remove a Server Dns Alias from cmdlet input parameters";
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft it looks like the parameter set names are still space-separate strings. I would recommend changing the names of them in this file to `RemoveByNameAndResourceGroup`, `RemoveByInputObject` and `RemoveByResourceId`

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+		public Model.AzureSqlServerDnsAliasModel InputObject { get; set; }
+
+		/// <summary>
+		/// Gets or sets the resource id of the server dns alias
+		/// </summary>
+		[Parameter(ParameterSetName = ResourceIdParameterSet,
+			Mandatory = true,
+			HelpMessage = "The resource id of Server Dns Alias object to remove")]
+		[ValidateNotNullOrEmpty]
+		public string ResourceId { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft `ResourceId` needs to have the `ValueFromPipelineByPropertyName = true` property

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+		[ValidateNotNullOrEmpty]
+		public override string ServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletInputParameterSet,
+			Mandatory = true,
+			Position = 0,
+			ValueFromPipelineByPropertyName = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs]: @maboja-msft please remove the `ValueFromPipelineByPropertyName` property from `ResourceGroupName`

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+		public static string ServerDnsAliasIdTemplate = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Sql/servers/{2}/dnsAliases/{3}";
+		
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server Dns Alias
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The Azure Sql Server Dns Alias name.")]
+		[Alias("DnsAliasName")]
+		[ValidateNotNullOrEmpty]
+		public string Name { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: @maboja-msft I can't remember if we discussed it in the review, but should this cmdlet have the `InputObject` and `ResourceId` parameter sets?

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+			HelpMessage = "The name of Azure Sql Server to which alias should point.")]
+		[ValidateNotNullOrEmpty]
+		public string TargetServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(Mandatory = true,
+			Position = 0,
+			ValueFromPipelineByPropertyName = true,
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: @maboja-msft if there's no `ResourceId` parameter set for this cmdlet, then we should have `Name` also have the `ValueFromPipelineByPropertyName` property

File: src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md [hunk 25]
```diff
@@ -0,0 +1,153 @@
...
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Name
+The Azure Sql Server DNS Alias name.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Get-AzureRmSqlServerDnsAlias.md]: @maboja-msft another pesky trailing ` ```yaml ` string :relaxed:

File: src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlServerDnsAlias.md [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Name
+The Azure Sql Server Dns Alias name.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlServerDnsAlias.md]: @maboja-msft same comment

File: src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlServerDnsAlias.md [hunk 27]
```diff
@@ -0,0 +1,185 @@
...
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Name
+The Azure Sql Server Dns Alias name.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlServerDnsAlias.md]: @maboja-msft same comment

File: src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs (hunk 23 above)
Comment (maboja-msft) [reply src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs]: Yes, we discussed that and decided we are not supporting piping for set command

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/103/

**Pull request merged**